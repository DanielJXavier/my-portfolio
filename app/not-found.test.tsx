import "tests/matchMedia.mock";

import "@testing-library/jest-dom";

import { render } from "@testing-library/react";

import NotFound from "./not-found";

describe("NotFound page", () => {
  it("renders the page", () => {
    const { container } = render(<NotFound />);

    expect(container).toMatchSnapshot();
  });

  it("renders the page with dark theme", () => {
    Object.defineProperty(window, "matchMedia", {
      value: jest.fn().mockImplementation(() => ({
        matches: true,
      })),
    });

    render(<NotFound />);

    expect(document.documentElement.className).toContain("dark");
  });
});
