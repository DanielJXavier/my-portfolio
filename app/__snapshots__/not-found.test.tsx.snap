// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound page renders the page 1`] = `
<div>
  <main
    class="main-container relative pt-32 flex justify-center"
  >
    <div
      class="absolute w-[320px] sm:w-[400px] lg:w-[450px] h-[265px] sm:h-[365px] lg:h-[432px]"
    >
      <div
        class="absolute sm:top-8 lg:top-10 flex flex-col gap-y-7 lg:gap-y-10 z-10"
      >
        <div>
          <p
            class="text-lg sm:text-xl lg:text-2xl"
          >
            Ooops!
          </p>
          <p
            class="text-xs sm:text-sm lg:text-base"
          >
            There is no page here.
          </p>
        </div>
        <a
          class="text-sm sm:text-base lg:text-lg font-medium underline hover:text-secondary"
          href="/en"
        >
          Let's see a real page!
        </a>
      </div>
      <div
        class="absolute top-0 right-0 bottom-0 left-32 z-0"
      >
        <img
          alt="Photo of the author (Daniel Xavier) indicating a 404 error with his finger"
          class="right-0 object-contain drop-shadow-shaped grayscale"
          data-nimg="fill"
          decoding="async"
          fetchpriority="high"
          sizes="50vw, (min-width: 640px): 50vw, (min-width: 1024px): 33vw"
          src="/_next/image?url=%2Fimages%2F404.png&w=3840&q=75"
          srcset="/_next/image?url=%2Fimages%2F404.png&w=256&q=75 256w, /_next/image?url=%2Fimages%2F404.png&w=384&q=75 384w, /_next/image?url=%2Fimages%2F404.png&w=640&q=75 640w, /_next/image?url=%2Fimages%2F404.png&w=750&q=75 750w, /_next/image?url=%2Fimages%2F404.png&w=828&q=75 828w, /_next/image?url=%2Fimages%2F404.png&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2F404.png&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2F404.png&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2F404.png&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2F404.png&w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
        />
      </div>
    </div>
  </main>
</div>
`;
